lib/phoenix/transports/long_poll.ex

```diff
@@ -261,18 +190,24 @@ defmodule Phoenix.Transports.LongPoll do
     Phoenix.Token.verify(endpoint, Atom.to_string(endpoint.__pubsub_server__), signed, opts[:crypto])
   end
 
-  defp status_json(conn, data) do
-    status = Plug.Conn.Status.code(conn.status || 200)
-    data   = Map.put(data, :status, status)
-    conn
-    |> put_status(200)
-    |> Phoenix.Controller.json(data)
+  defp status_json(conn) do
+    status = conn.status || 200
+    send_json(conn, [{"status", Integer.to_string(status)}])
+  end
+
+  defp status_token_messages_json(conn, token, messages) do
+    status = conn.status || 200
+    messages = [?[, Enum.intersperse(messages, ?,), ?]]
+    token = Phoenix.json_library().encode_to_iodata!(token)
+    data = [{"status", Integer.to_string(status)}, {"token", token}, {"messages", messages}]
+    send_json(conn, data)
   end
 
-  defp code_reload(conn, opts, endpoint) do
-    reload? = Keyword.get(opts, :code_reloader, endpoint.config(:code_reloader))
-    if reload?, do: Phoenix.CodeReloader.reload!(endpoint)
+  defp send_json(conn, json) do
+    data = for {k, v} <- json, do: [?", k, ?", ?:, v]
```

Comment:
Are you sure the data does not contain anything that has to be escaped in JSON? Why not use the encoder?

Reply:
Yes, I am because all of the calls happen on this module.

We cannot use the encoder because the data that comes from the socket is already encoded. Otherwise we would need to encode it twice.